scripts/build_energy_totals.py

```diff
@@ -567,15 +571,53 @@ def build_idees(countries: List[str]) -> pd.DataFrame:
         names=["country", "year"],
     )
 
+    # clean up dataframe
+    years = np.arange(2000, 2022)
+    totals = totals[totals.index.get_level_values(1).isin(years)]
+
     # efficiency kgoe/100km -> ktoe/100km so that after conversion TWh/100km
     totals.loc[:, "passenger car efficiency"] /= 1e6
     # convert ktoe to TWh
     exclude = totals.columns.str.fullmatch("passenger cars")
+    totals = totals.copy()
     totals.loc[:, ~exclude] *= 11.63 / 1e3
 
     return totals
 
 
+def fill_missing_years(fill_values: pd.Series) -> pd.Series:
+    """
+    Fill missing years for some countries by first using forward fill (ffill)
+    and then backward fill (bfill).
+
+    Parameters
+    ----------
+    fill_values : pd.Series
+        A pandas Series with a MultiIndex (levels: country and year) representing
+        energy values, where some values may be zero and need to be filled.
+
+    Returns
+    -------
+    pd.Series
+        A pandas Series with zero values replaced by the forward-filled and
+        backward-filled values of the corresponding country.
+
+    Notes
+    -----
+    - The function groups the data by the 'country' level and performs forward fill
+      and backward fill to fill zero values.
+    - Zero values in the original Series are replaced by the ffilled and bfilled
+      value of their respective country group.
+    """
+    # Replace zero values with NaN for correct filling
+    fill_values = fill_values.replace(0, pd.NA)
```

Comment:
In principle, this can go wrong when 0 is the actual value for this entry.

When reading in the Excel files, do we maybe need to specify special NaN values, i.e. `pd.read_excel(...., na_values="X")`?